var/spack/repos/builtin/packages/py-pyarrow/package.py

```diff
@@ -13,8 +13,11 @@ class PyPyarrow(PythonPackage):
     """
 
     homepage = "http://arrow.apache.org"
-    url      = "https://pypi.org/packages/source/p/pyarrow/pyarrow-0.9.0.tar.gz"
+    url = 'https://www.apache.org/dyn/closer.cgi/arrow/arrow-0.15.0/apache-arrow-0.15.0.tar.gz'
 
+    version('0.15.1', sha256='ab1c0d371a10b615eccfcead71bb79832245d788f4834cc6b278c03c3872d593')
+    version('0.15.0', sha256='d1072d8c4bf9166949f4b722a89350a88b7c8912f51642a5d52283448acdfd58')
+    version('0.14.1', sha256='69d9de9ec60a3080543b28a5334dbaf892ca34235b8bd8f8c1c01a33253926c1')
     version('0.12.1', sha256='10db6e486c918c3af999d0114a22d92770687e3a6607ea3f14e6748854824c2a')
     version('0.11.0', sha256='07a6fd71c5d7440f2c42383dd2c5daa12d7f0a012f1e88288ed08a247032aead')
     version('0.9.0', sha256='7db8ce2f0eff5a00d6da918ce9f9cfec265e13f8a119b4adb1595e5b19fd6242')
```

Comment:
If you're changing the download URL for all versions, you may need to change the checksums too.

Reply:
see the top post :) Unfortunately I failed to fetch these (curl finds html)

Reply:
Why not stick with downloading from PyPI then?

Reply:
because not all versions are present there. How would I set `build_directory` version-dependent?

Reply:
> because not all versions are present there.

They seem to be present to me: https://pypi.org/project/pyarrow/#history

Reply:
mh, when I was looking there were 14.1 and 15.1 only available from the apache website, seems I hit the small timeframe that it took them to push to pypi -> going back to pypi sources

Reply:
You could just drop 0.14.1 and 0.15.0, it looks like they switched back to providing tarballs for 0.15.1. Or you could switch everything to download from Apache/GitHub, in which case you need to change the checksums.